# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1296105340 18000
#      Thu Jan 27 00:15:40 2011 -0500
# Branch release-3-4-x
# Node ID 4f8069727e9f788c0e781b640f30fb6d8c8553b3
# Parent  7cc26ca98b00c6c72fb94fb80617b32c50d39866
scripts/Makefile.am (check-m-semicolon): new rule.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2011-01-27  John W. Eaton  <jwe@octave.org>
+
+	* Makefile.am (check-missing-semicolon): New rule.
+
 2011-01-26  Rik  <octave@nomad.inbox5.com>
 
 	* scripts/general/interp1.m, scripts/general/interp2.m,
 	scripts/general/interp3.m, scripts/general/interpn.m: Use em-dash
 	rather than en-dash.
 	* scripts/miscellaneous/mkoctfile.m: Use @code to prevent option
 	turning to en-dash.
 
diff --git a/scripts/Makefile.am b/scripts/Makefile.am
--- a/scripts/Makefile.am
+++ b/scripts/Makefile.am
@@ -347,16 +347,24 @@ check-m-sources:
 	    echo "$$missing: not listed in SOURCES"; \
 	  fi; \
 	done; \
 	if test -z "$$missing"; then \
 	  echo "yes"; \
 	fi
 .PHONY: check-m-sources
 
+check-missing-semicolon:
+	@echo "checking for missing semicolons in .m files..."
+	( echo "warning on Octave:missing-semicolon;"; \
+	  for m in $(addprefix $(srcdir)/, $(FCN_FILES)) $(GEN_FCN_FILES); do \
+	    echo "source ('$$m');"; \
+	  done ) | ../run-octave -qf
+.PHONY: check-missing-semicolon
+
 install-data-local: install-startup-files install-pkg-add
 
 uninstall-local: uninstall-startup-files uninstall-pkg-add
 
 install-startup-files:
 	$(MKDIR_P) $(DESTDIR)$(fcnfiledir)/startup
 	if test -f $(DESTDIR)$(fcnfiledir)/startup/octaverc; then true; \
 	else \
